UITests/UITests.App/MainTestHost.xaml.cs

```diff
@@ -0,0 +1,118 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Mvvm.Messaging;
+using Microsoft.Toolkit.Uwp.Extensions;
+using UITests.App.Pages;
+using Windows.System;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media.Animation;
+
+namespace UITests.App
+{
+    /// <summary>
+    /// MainPage hosting all other test pages.
+    /// </summary>
+    public sealed partial class MainTestHost : IRecipient<RequestPageMessage>
+    {
+        private DispatcherQueue _queue;
+
+        private Assembly _executingAssembly = Assembly.GetExecutingAssembly();
+
+        private TaskCompletionSource<bool> _loadingStateTask;
+
+        public MainTestHost()
+        {
+            InitializeComponent();
+
+            WeakReferenceMessenger.Default.RegisterAll(this);
```

Comment:
If we have a single recipient here, we can individually register for the message type we need.
That's more efficient as it won't require reflection and dynamic code generation:

```suggestion
            WeakReferenceMessenger.Default.Register<RequestPageMessage>(this);
```